### PR TITLE
Convert receive_date and renewal_date on membership forms to use date…

### PR DESCRIPTION
**Pull request by seamuslee001**
…picker rather than jcalendar

Overview
----------------------------------------
This converts the receive_date and the renewal_date fields in Membership forms to use the datepicker format rather than jcalander

Before
----------------------------------------
legacy jcalendar used

After
----------------------------------------
modern datepicker used

ping @eileenmcnaughton @mattwire @monishdeb

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@seamuslee001 jenkins needs some love

**Comment by eileenmcnaughton** [comment]
```
PHP Files:
 * CRM/Member/Form/Membership.php
 * CRM/Member/Form/MembershipRenewal.php
Skip Files:
 * templates/CRM/Member/Form/MembershipCommon.tpl
 * templates/CRM/Member/Form/MembershipRenewal.tpl
===============================[ php -l ]===============================
PHP Parse error:  syntax error, unexpected ';' in CRM/Member/Form/MembershipRenewal.php on line 492

Parse error: syntax error, unexpected ';' in CRM/Member/Form/MembershipRenewal.php on line 492
```

File: CRM/Member/Form/MembershipRenewal.php [hunk 1]
```diff
@@ -502,8 +489,7 @@ protected function submit() {
     $this->storeContactFields($this->_params);
     $this->beginPostProcess();
     $now = CRM_Utils_Date::getToday(NULL, 'YmdHis');
-    $this->convertDateFieldsToMySQL($this->_params);
-    $this->assign('receive_date', $this->_params['receive_date']);
+    $this->assign('receive_date', CRM_Utils_Array::value('recieve_date', $this->_params, date('Y-m-d H:i:s'));
```
Comment (eileenmcnaughton) [inline CRM/Member/Form/MembershipRenewal.php]: typo + jenkins hates this line too

File: CRM/Member/Form/MembershipRenewal.php [hunk 2]
```diff
@@ -502,8 +489,7 @@ protected function submit() {
     $this->storeContactFields($this->_params);
     $this->beginPostProcess();
     $now = CRM_Utils_Date::getToday(NULL, 'YmdHis');
-    $this->convertDateFieldsToMySQL($this->_params);
-    $this->assign('receive_date', $this->_params['receive_date']);
+    $this->assign('receive_date', CRM_Utils_Array::value('recieve_date', $this->_params, date('Y-m-d H:i:s')));
```
Comment (eileenmcnaughton) [inline CRM/Member/Form/MembershipRenewal.php]: less jenkins hate, stil typo

Comment (seamuslee001) [reply CRM/Member/Form/MembershipRenewal.php]: hopefully fixed now

File: CRM/Member/Form/MembershipRenewal.php [hunk 3]
```diff
@@ -502,8 +489,7 @@ protected function submit() {
     $this->storeContactFields($this->_params);
     $this->beginPostProcess();
     $now = CRM_Utils_Date::getToday(NULL, 'YmdHis');
-    $this->convertDateFieldsToMySQL($this->_params);
-    $this->assign('receive_date', $this->_params['receive_date']);
+    $this->assign('receive_date', CRM_Utils_Array::value('receive_date', $this->_params, date('Y-m-d H:i:s')));
```
Comment (eileenmcnaughton) [inline CRM/Member/Form/MembershipRenewal.php]: recEIving you load & clear now

File: CRM/Member/Form/Membership.php [hunk 4]
```diff
@@ -1618,6 +1599,10 @@ public function submit() {
             $membershipTypeValues[$memType]['relate_contribution_id'] = $relateContribution;
           }
 
+          // @todo figure out why recieve_date isn't being set right here.
```
Comment (seamuslee001) [inline CRM/Member/Form/Membership.php]: @eileenmcnaughton WHYY?????

**Comment by eileenmcnaughton** [comment]
@seamuslee001 this seemed to work for me but your comment suggests an issue?

**Comment by seamuslee001** [comment]
@eileenmcnaughton when i was running `CRM_Member_Form_MembershipTest.testSubmitUpdateMembershipFromPartiallyPaid` it seemed to fail because for some reason receive_date was not getting set hence why i added in the if (empty and the todo

**Comment by eileenmcnaughton** [comment]
@seamuslee001 OK I think that's non-blocking it seemed to save it for me

**Pull request merged**